flutter/realm_flutter/tests/test_driver/realm_test.dart

```diff
@@ -19,32 +19,37 @@ import '../test/session_test.dart' as session_test;
 
 Future<String> main(List<String> args) async {
   final Completer<String> completer = Completer<String>();
-  final List<String> failedTests = [];
-
-  await configuration_test.main(args);
-  await realm_tests.main(args);
-  await realm_object_tests.main(args);
-  await list_tests.main(args);
-  await results_tests.main(args);
-  await credentials_tests.main(args);
-  await app_tests.main(args);
-  await user_tests.main(args);
-  await subscription_test.main(args);
-  await session_test.main(args);
-
-  tearDown(() {
-    if (Invoker.current?.liveTest.state.result == test_api.Result.error || Invoker.current?.liveTest.state.result == test_api.Result.failure) {
-      failedTests.add(Invoker.current!.liveTest.individualName);
-    }
-  });
-
-  tearDownAll(() {
-    if (failedTests.isNotEmpty) {
-      completer.complete(failedTests.join('\n'));
-    } else {
-      completer.complete('');
-    }
-  });
+
+  try {
+    final List<String> failedTests = [];
+
+    await configuration_test.main(args);
+    await realm_tests.main(args);
+    await realm_object_tests.main(args);
+    await list_tests.main(args);
+    await results_tests.main(args);
+    await credentials_tests.main(args);
+    await app_tests.main(args);
+    await user_tests.main(args);
+    await subscription_test.main(args);
+    await session_test.main(args);
+
+    tearDown(() {
+      if (Invoker.current?.liveTest.state.result == test_api.Result.error || Invoker.current?.liveTest.state.result == test_api.Result.failure) {
+        failedTests.add(Invoker.current!.liveTest.individualName);
+      }
+    });
+
+    tearDownAll(() {
+      if (failedTests.isNotEmpty) {
+        completer.complete(failedTests.join('\n'));
+      } else {
+        completer.complete('');
+      }
+    });
+  } catch (e) {
+    completer.complete(e.toString());
```

Comment:
Should it be:
```suggestion
    completer.completeError(e.toString());
```

Reply:
no we don't need to do that, we need the string result